src/oisst/State/State.cc

```diff
@@ -5,120 +5,366 @@
 
+#include "atlas/field.h"
+#include "atlas/array.h"
+#include "atlas/option.h"
+
 #include "oops/base/Variables.h"
 #include "oops/util/abor1_cpp.h"
+#include "oops/util/Logger.h"
 
 #include "ufo/GeoVaLs.h"
 #include "ufo/Locations.h"
 
+// using namespace netCDF;
+using atlas::array::make_view;
+using atlas::option::name;
+
+
 namespace oisst {
 
 // ----------------------------------------------------------------------------
 
   State::State(const Geometry & geom, const eckit::Configuration & conf)
     : geom_(new Geometry(geom)), time_(conf.getString("date")),
       vars_(conf, "state variables") {
-    util::abor1_cpp("State::State() needs to be implemented.",
-                     __FILE__, __LINE__);
+
+    if (vars_.size() != 1) {
+      util::abor1_cpp("State::State() needs to be implemented.",
+                       __FILE__, __LINE__);
+    }
+
+    const int SIZE = geom_->atlasFunctionSpace()->size();
+
+    atlasFieldSet_.reset(new atlas::FieldSet());
+    for (int i = 0; i < vars_.size(); i++) {
+      std::string var = vars_[i];
+      atlas::Field fld = geom_->atlasFunctionSpace()->createField<float>(
+                         name(var));
+      // Initialize to 0.0 or 300.0
+      auto fd = make_view<float, 1>(fld);
+      fd.assign(300.0);  // norm is 300., 0.0 if init to 0.0;
+
+      atlasFieldSet_->add(fld);
+    }
+
+    // check and read in data if it has filename
+    if (conf.has("filename")) {
+      read(conf);
+//    double norm1 = norm();
+//    std::cout << "State::State(), after read file, norm1=" << norm1
+//              << std::endl;
+    }
   }
 
 // ----------------------------------------------------------------------------
 
   State::State(const Geometry & geom, const oops::Variables & vars,
                const util::DateTime & time)
     : geom_(new Geometry(geom)), time_(time), vars_(vars) {
-    util::abor1_cpp("State::State() needs to be implemented.",
-                    __FILE__, __LINE__);
+
+    if (vars_.size() != 1) {
+      util::abor1_cpp("State::State() needs to be implemented.",
+                      __FILE__, __LINE__);
+    }
+
+    const int SIZE = geom_->atlasFunctionSpace()->size();
+
+    atlasFieldSet_.reset(new atlas::FieldSet());
+    for (int i = 0; i < vars_.size(); i++) {
+      std::string var = vars_[i];
+      atlas::Field fld = geom_->atlasFunctionSpace()->createField<float>(
+                         name(var));
+      // Initialize to 300.0, will fail the test: EXPECT(xx3.norm()==0); so 0.0
+      auto fd = make_view<float, 1>(fld);
+      fd.assign(0.0);
+
+      atlasFieldSet_->add(fld);
+    }
   }
 
 // ----------------------------------------------------------------------------
 
   State::State(const Geometry & geom, const State & other)
     : geom_(new Geometry(geom)), time_(other.time_), vars_(other.vars_) {
     // Change state resolution
-    util::abor1_cpp("State::State() needs to be implemented.",
-                    __FILE__, __LINE__);
+
```

Comment:
This constructor is used to interpolate the state from one resolution to another. We don't need that any time soon I think, so we can leave it the way it is. But just something to keep in mind in the future if we do in fact need resolution change of state.